sql/core/src/test/scala/org/apache/spark/sql/InjectRuntimeFilterSuite.scala

```diff
@@ -257,6 +257,11 @@ class InjectRuntimeFilterSuite extends QueryTest with SQLTestUtils with SharedSp
         val normalizedDisabled = normalizePlan(normalizeExprIds(planDisabled))
         ensureLeftSemiJoinExists(planEnabled)
         assert(normalizedEnabled != normalizedDisabled)
+        val agg = planEnabled.collect {
+          case Join(_, agg: Aggregate, LeftSemi, _, _) => agg
+        }
+        assert(agg.size == 1)
+        assert(agg.head.fastEquals(ColumnPruning(agg.head)))
```

Comment:
this test can pass without this pr because
https://github.com/apache/spark/blob/38897b1ce96a1e24629875619c165b4fd1fa2d8f/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala#L68-L79

I think it is just a coincidence since we converted the subquery to left semi join ..